### PR TITLE
reverse-string: add argument to initial solution

### PR DESCRIPTION
**Pull request by bmkiefer**
I went to solve this exercise on exercism.io and noticed that I forgot the initial argument on the implementation. I figured this may be confusing to learners as I was confused when I ran the test suite after pulling it down.

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/xjava/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

File: exercises/reverse-string/src/main/java/ReverseString.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class ReverseString {
   
-    String reverse() {
+    String reverse(String input) {
```
Comment (stkent) [inline exercises/reverse-string/src/main/java/ReverseString.java]: Nice catch! Let's go with `String inputString` to match the reference solution :)

**Comment by sjwarner-bp** [comment]
Oops! Didn't see this one - thanks for pointing that out @bmkiefer, I'll be more careful in future!

**Comment by bmkiefer** [comment]
No worries! Glad to see stuff getting done

**Comment by bmkiefer** [comment]
Weird another unclear CI break.

2017/12/07 18:14:58 error parsing API response - invalid character '<' looking for beginning of value

**Comment by FridaTveit** [comment]
I've don't think I've seen an error like that before. @exercism/java, any ideas?

**Comment by sjwarner-bp** [comment]
I'll take a look at it when I'm back at my computer! ☺️

**Comment by Smarticles101** [comment]
Looks like a possible bug with exercism-cli to me...

The actual test for `roman-numerals` did pass however:
```
+echo ==================================================
==================================================
+echo '3 of 90 -- reverse-string'
3 of 90 -- reverse-string
+echo ==================================================
==================================================
+./exercism --config .journey-test.exercism.json fetch java reverse-string
127.0.0.1 - - [07/Dec/2017:18:05:36 +0000] "GET /v2/exercises/java/reverse-string HTTP/1.1" 200 2354 0.0077
Not Submitted:        1 problem
java (Reverse String) /home/travis/build/exercism/java/build/exercism/java/reverse-string
New:                  1 problem
java (Reverse String) /home/travis/build/exercism/java/build/exercism/java/reverse-string
unchanged: 0, updated: 0, new: 1
+cp -R -H /home/travis/build/exercism/java/exercises/reverse-string/.meta/src/reference/java/ReverseString.java /home/travis/build/exercism/java/build/exercism/java/reverse-string/src/main/java/
+pushd /home/travis/build/exercism/java/build/exercism/java/reverse-string
~/build/exercism/java/build/exercism/java/reverse-string ~/build/exercism/java/build/exercism ~/build/exercism/java
+/home/travis/build/exercism/java/gradlew compileTestJava
Picked up _JAVA_OPTIONS: -Xmx2048m -Xms512m
:compileJava
:processResources NO-SOURCE
:classes
:compileTestJava
BUILD SUCCESSFUL in 5s
2 actionable tasks: 2 executed
++find . -name '*Test.java'
+for testfile in '`find . -name "*Test.${TRACK_SRC_EXT}"`'
+sed 's/@Ignore\(\(.*\)\)\{0,1\}//' ./src/test/java/ReverseStringTest.java
+mv /tmp/journey-test.sh-unignore_all_tests.txt ./src/test/java/ReverseStringTest.java
+/home/travis/build/exercism/java/gradlew test
Picked up _JAVA_OPTIONS: -Xmx2048m -Xms512m
:compileJava UP-TO-DATE
:processResources NO-SOURCE
:classes UP-TO-DATE
:compileTestJava
:processTestResources NO-SOURCE
:testClasses
:testPicked up _JAVA_OPTIONS: -Xmx2048m -Xms512m
ReverseStringTest > testAWord PASSED
ReverseStringTest > testAnEmptyString PASSED
ReverseStringTest > testACapitalizedWord PASSED
ReverseStringTest > testASentenceWithPunctuation PASSED
ReverseStringTest > testAPalindrome PASSED
BUILD SUCCESSFUL in 6s
3 actionable tasks: 2 executed, 1 up-to-date
+popd
~/build/exercism/java/build/exercism ~/build/exercism/java
+current_exercise_number=4
+for exercise in '$exercises'
+echo -e '\n\n'
```

**Comment by sjwarner-bp** [comment]
I've had a look, and the build log seems very odd to say the least.

```
+echo ==================================================
==================================================
+echo '46 of 90 -- roman-numerals'
46 of 90 -- roman-numerals
+echo ==================================================
==================================================
+./exercism --config .journey-test.exercism.json fetch java roman-numerals
127.0.0.1 - - [07/Dec/2017:18:14:28 +0000] "GET /v2/exercises/java/roman-numerals HTTP/1.1" 200 3912 0.0044
2017/12/07 18:14:58 error parsing API response - invalid character '<' looking for beginning of value
+on_exit
+echo '>>> on_exit()'
>>> on_exit()
```

**Comment by bmkiefer** [comment]
hey, @stkent would you mind approving these changes so this doesn't have to sit open? Thanks!

**Pull request merged**